roachtest: add admission control variations to test

andrewbaptist opened the pull request:
Add options to specify different admission control modes to the variations.

Fixes: #133412

Release note: None

Commit:
roachtest: add admission control variations to perturbation/* test

Add options to specify different admission control modes to the
perturbation variations. This only impacts the
perturbation/metamorphic/* variations, however it allows running the
other tests with different AC modes.

Fixes: #133412

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/135024)
<!-- Reviewable:end -->

kvoli reviewed (commented):
> Reviewed 1 of 1 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/135024)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @andrewbaptist, @herkolategan, and @srosenberg)

___
*[`-- commits` line 2 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/135024#-OBfHW-10Zm9IDyeTu5X:-OBfHW-10Zm9IDyeTu5Y:b-tbcauq):*
Specify the perturbation package.
___
*[`-- commits` line 4 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/135024#-OBfH_CGAYQMy6SX9NPz:-OBfH_CGAYQMy6SX9NQ-:b-ixh6o2):*
Likewise here.
___
*[`pkg/cmd/roachtest/tests/perturbation/framework.go` line 170 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/135024#-OBfHtnQ3NxS4GUhZ55O:-OBfHtnQ3NxS4GUhZ55P:bnwv2d9) ([raw file](https://github.com/cockroachdb/cockroach/blob/1dd78ac9cbcf1cf6b95ab75ec6fec36bc5cd110e/pkg/cmd/roachtest/tests/perturbation/framework.go#L170)):*
> ```Go
> 	case elasticOnlyBoth:
> 		return map[string]string{
> 			"admission.kv.bulk_only.enabled": "true",
> ```

What is `bulk_only` for? Are we intending to run it automatically via CI? Generally, I was thinking of something much simpler here, just apply_to_elastic and apply_to_all.
___
*[`pkg/cmd/roachtest/tests/perturbation/framework.go` line 281 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/135024#-OBfI9W33dPLKGUCEmuU:-OBfI9W33dPLKGUCEmuV:b-cpvsfv) ([raw file](https://github.com/cockroachdb/cockroach/blob/1dd78ac9cbcf1cf6b95ab75ec6fec36bc5cd110e/pkg/cmd/roachtest/tests/perturbation/framework.go#L281)):*
> ```Go
> 
> // finish completes initialization of the variations.
> func (v variations) finish() variations {
> ```

nit: finishing the setup might be better named `finishSetup`, or `completeSetup`.


<!-- Sent from Reviewable.io -->

andrewbaptist reviewed (commented):
TFTR! changes made!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/135024#-:-OBgU3WA9nBmWptibLNL:b-35g6rp)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @herkolategan, @kvoli, and @srosenberg)

___
*[`-- commits` line 2 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/135024#-OBfHW-10Zm9IDyeTu5X:-OBgU5vVBsh4vOooUxv0:bb6un1j):*
<details><summary><i>Previously, kvoli (Austen) wrote…</i></summary><blockquote>

Specify the perturbation package.
</blockquote></details>

Done
___
*[`-- commits` line 4 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/135024#-OBfH_CGAYQMy6SX9NPz:-OBgU77xCtkqoekAOj4r:bb6un1j):*
<details><summary><i>Previously, kvoli (Austen) wrote…</i></summary><blockquote>

Likewise here.
</blockquote></details>

Done
___
*[`pkg/cmd/roachtest/tests/perturbation/framework.go` line 170 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/135024#-OBfHtnQ3NxS4GUhZ55O:-OBgUAJMB-9DI6rSNcEf:bvk1nld) ([raw file](https://github.com/cockroachdb/cockroach/blob/1dd78ac9cbcf1cf6b95ab75ec6fec36bc5cd110e/pkg/cmd/roachtest/tests/perturbation/framework.go#L170)):*
<details><summary><i>Previously, kvoli (Austen) wrote…</i></summary><blockquote>

What is `bulk_only` for? Are we intending to run it automatically via CI? Generally, I was thinking of something much simpler here, just apply_to_elastic and apply_to_all.
</blockquote></details>

The options above `admissionControlOptions` are the three modes we will run in metamorphic tests. We have a number of customers still running with bulk only mode and it is useful to make sure we don't break that. I think we can remove that in the future once we roll out apply to all as the default, but its useful to have this as a test for now.
___
*[`pkg/cmd/roachtest/tests/perturbation/framework.go` line 281 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/135024#-OBfI9W33dPLKGUCEmuU:-OBgUStAAge4BEJCvdZA:bgn9dv7) ([raw file](https://github.com/cockroachdb/cockroach/blob/1dd78ac9cbcf1cf6b95ab75ec6fec36bc5cd110e/pkg/cmd/roachtest/tests/perturbation/framework.go#L281)):*
<details><summary><i>Previously, kvoli (Austen) wrote…</i></summary><blockquote>

nit: finishing the setup might be better named `finishSetup`, or `completeSetup`.
</blockquote></details>

Changed to finishSetup. I also cleaned up the code to correctly return an error if the env variable is set, but the option isn't correctly named.


<!-- Sent from Reviewable.io -->

Commit:
roachtest: add overrides to perturbation/* tests

Allow the user to override perturbations by specifying the environment
variable `PERTURBATION_OVERRIDE`. This applies to all perturbation
tests, but is useful for reproducing specific metamorphic failures where
it is unclear which parameter change caused the difference.

An example of using an override to modify the test block size:

```
PERTURBATION_OVERRIDE=blockSize=1024 roachtest run
perturbation/full/addNode
```

Epic: none

Release note: None

Commit:
roachtest: display cluster settings in perturbation/* tests

It is useful to know what cluster settings are being applied in the
tests. This change logs them before applying them.

Epic: none

Release note: None

Commit:
roachtest: add finishSetup method to perturbation/* tests

This commit creates a finishSetup method which is called at the end of
the setup phase of perturbation tests. The method can be used to do any
application of variables at the end of the test.

Epic: none

Release note: None

andrewbaptist commented:
TFTR! We'll starting geting metamorphic variations for apply_to_all this weekend.

bors r=kvoli
